File: src/GroundZero.Application/Features/WorkoutPlans/DTOs/AddWorkoutExerciseRequest.cs
namespace GroundZero.Application.Features.WorkoutPlans.DTOs;

public class AddWorkoutExerciseRequest
{
    public int ExerciseId { get; set; }
    public int Sets { get; set; }
    public int Reps { get; set; }
    public decimal? Weight { get; set; }
    public int? RestSeconds { get; set; }
    public int OrderIndex { get; set; }
}

File: groundzero_mobile/lib/features/workouts/models/add_workout_exercise_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'add_workout_exercise_request.freezed.dart';
part 'add_workout_exercise_request.g.dart';

@freezed
class AddWorkoutExerciseRequest with _$AddWorkoutExerciseRequest {
  const factory AddWorkoutExerciseRequest({
    required int exerciseId,
    required int sets,
    required int reps,
    double? weight,
    int? restSeconds,
    required int orderIndex,
  }) = _AddWorkoutExerciseRequest;

  factory AddWorkoutExerciseRequest.fromJson(Map<String, dynamic> json) =>
      _$AddWorkoutExerciseRequestFromJson(json);
}
